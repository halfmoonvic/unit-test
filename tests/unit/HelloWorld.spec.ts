import { shallowMount } from '@vue/test-utils';
import HelloWorld from '@/components/HelloWorld.vue';
import Vue from 'vue';

describe('HelloWorld.vue', () => {
  it('通过手写 html 等来测试', () => {
    const root = document.createElement('div');
    root.className = 'root';
    document.body.appendChild(root);
    new Vue({
      render: h =>
        h(HelloWorld, {
          props: {
            msg: 'msg'
          }
        })
    }).$mount('.root');
    expect(document.getElementsByClassName('hello').length).toBe(1);
  });

  it('renders props.msg when passed', () => {
    const msg = 'new message abc';
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg }
    });
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper).toMatchSnapshot();
  });
});
