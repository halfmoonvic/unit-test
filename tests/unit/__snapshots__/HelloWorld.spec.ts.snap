// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders props.msg when passed 1`] = `
<div class="hello">
  <h1>new message abc</h1>
</div>
`;
